Add Thermal Analyzer document

mwjohnson56 opened the pull request:
This adds a thermal analyzer document

mwjohnson56 committed:
add thermal analyzer doc

mwjohnson56 committed:
add links

elsevers reviewed (changes requested):
This PR appears to be doing more than just adding a thermal analyzer. It is also modifying the Analytic Machine Design tutorial in a way that conflicts with the `develop` branch. Can we get clean PRs for each of these?

(One PR for the thermal analyzer and one PR for the changes to the Analytic Machine Design tutorial)

mwjohnson56 committed:
remove update to links

mwjohnson56 committed:
Merge branch 'develop' into add-therm-ana-doc

mwjohnson56 commented:
I had added the links to the ``Analytical machine design`` tutorial in this PR as well, but for some reason it threw a merge conflict. I have removed those changes from this PR and will push it on a separate branch. 

In addition, I have fixed a small bug in the thermal analyzer code, the number of nodes was hard coded as 33, I have changed this now to reference a ``N_nodes`` field in the ``problem``. This wasn't throwing any errors and wasn't affecting any math, it just placed a cap on the number of nodes a future problem could use.
